datasource.go

```diff
@@ -0,0 +1,120 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"golang.org/x/net/context"
+
+	"github.com/grafana/grafana-plugin-model/go/datasource"
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	plugin "github.com/hashicorp/go-plugin"
+	"golang.org/x/net/context/ctxhttp"
+)
+
+var httpClient = &http.Client{}
+
+type ClickhouseDatasource struct {
+	plugin.NetRPCUnsupportedPlugin
+}
+
+func (t *ClickhouseDatasource) Query(ctx context.Context, req *datasource.DatasourceRequest) (*datasource.DatasourceResponse, error) {
+	refId := req.Queries[0].RefId
+	modelJson, err := simplejson.NewJson([]byte(req.Queries[0].ModelJson))
+	if err != nil {
+		return nil, err
+	}
+	
+	query := modelJson.Get("rawQuery").MustString()
+	request, err := createRequest(req, query)
+	if err != nil {
+		return nil, err
+	}
+
+	response, err := ctxhttp.Do(ctx, httpClient, request)
+	if err != nil {
+		return nil, err
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("invalid status code. status: %v", response.Status)
+	}
+
+	defer response.Body.Close()
+
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return parseResponse(body, refId)
+}
+
+func createRequest(req *datasource.DatasourceRequest, query string) (*http.Request, error) {
+	escaped := url.QueryEscape(query + " FORMAT JSON")
+
+	// TODO support other datasource options
+
+	return http.NewRequest(http.MethodGet, fmt.Sprintf("%s/?query=%s", req.Datasource.Url, escaped), nil)
```

Comment:
could you implement authorization options described on frontend part of plugin?
it will be really unsuccessful and bad user experience here if we miss authorization options

Reply:
Yeah, sounds good, I will update this